### ISSUE TITLE
create a email based authentication in django

### ISSUE THREAD

**Issue by 7saikat7** [7saikat7/supply_chain#57]

**Comment by abhishak3** [comment]
I would like to take up this issue.

**Comment by ChucklesDroid** [comment]
can I work on this issue?

**Comment by VaibhavSinghDev** [comment]
Can I work on this issue?

**Comment by 7saikat7** [comment]
hey , it's done ?

**Comment by ChucklesDroid** [comment]
I ran into an issue for the past 3 days, I have commented the same in the channel just now.
So my form.is_valid() is returning false. I am making use of the custom user model and have made a custom form for Registration and Login. I understand that forms would need the mandatory fields to be filled and that by default username, email and password are required fields but I am initialising them in my forms through constructor to overcome this.

![image](https://user-images.githubusercontent.com/71250179/184461135-e68712ff-20b5-49b8-890c-22a07dad618f.png)

and this the backend 
![image](https://user-images.githubusercontent.com/71250179/184461157-00765d92-117a-4656-ae5c-addb3df8cbc8.png)

this login view
![image](https://user-images.githubusercontent.com/71250179/184461177-6707aad7-46ee-4e86-bebb-9330d9781a9d.png)
